### DISCUSSION TITLE
404 This page could not be found. Nextjs Link

### DISCUSSION THREAD

**Discussion by xts-bit** [vercel/next.js#49159] [Help]
### Summary

i am learning nextjs and i am trying to navigate to ab component in nextjs with the link component can anyone can please tell me how to do things properly currently its giving me 404 This page could not be found. error when I try to do that

Footer:

```
import React from 'react'
import Link from 'next/link'

export default function Footer() {
  return (
    <footer className="bg-black-800 py-12">
      <div className="container mx-auto px-4">
        <div className="flex flex-wrap justify-between">
          <nav className="w-full lg:w-auto lg:flex-1 lg:text-center flex flex-wrap justify-center">
            <Link href="/ab" className="footer-link mr-4">
              <span>Privacy Policy</span>
            </Link>
            <Link href="/terms" className="footer-link mr-4">
              <span>Terms of Service</span>
            </Link>
          </nav>
        </div>
      </div>
    </footer>
  )
}
```
Componet:

```
import React from 'react'

export default function ab() {
  return (
    <div>
        <h1>Hello From compoent</h1>
    </div>
  )
}
```

### Additional information

```js
404 This page could not be found
```

### Example
This is how my files and folder  structure look like image = https://i.stack.imgur.com/NrYFg.jpg

**Comment by icyJoseph** [comment]
Only the files in pages can be pointed to by `next/link`'s href prop. 

If you want to go to `/ab`, there must be a `pages/ab.js` file.

**Reply by xts-bit** [reply]
@icyJoseph is  there any different way i want to make the ab.js file in a different folder

**Reply by icyJoseph** [reply]
Not really... the routing system Next.js uses is the pages file system. You can define your `ab` page in pages though, and just import the component that runs the page into it.

**Reply by xts-bit** [reply]
@icyJoseph   can i create a index.js in pages directory in my nextjs project and import the ab.js file from another folder will it work that way?

**Reply by icyJoseph** [reply]
You mean importing your `ab` component into `pages/index.js`? No, there must be a `pages/ab.js` file.

You should also use first Uppercase for your component, like Ab or AB:

- https://legacy.reactjs.org/docs/jsx-in-depth.html#user-defined-components-must-be-capitalized
- https://github.com/kentcdodds/react-fundamentals/blob/main/src/exercise/04.md#2--using-a-custom-component-with-jsx

You'd have:

```jsx
// components/ab.js
// import React from 'react' - you don't need to import React anymore

export default function Ab() {
  return (
    <div>
        <h1>Hello From compoent</h1>
    </div>
  )
}

```

And then in the pages directory:

```js
// pages/ab.js
import { Ab } from 'components/ab';

const AbPage = (props) => <Ab {...props} />;

export default AbPage;

```

**Reply by Shoharab01** [reply]
@icyJoseph Hi,
what about the dynamic route?

**Comment by suhel1519** [comment]
i have created a page called buy.tsx in pages but it shows the same error

**Comment by ISHA50** [comment]
> ### Summary
> i am learning nextjs and i am trying to navigate to ab component in nextjs with the link component can anyone can please tell me how to do things properly currently its giving me 404 This page could not be found. error when I try to do that
> 
> Footer:
> 
> ```
> import React from 'react'
> import Link from 'next/link'
> 
> export default function Footer() {
>   return (
>     <footer className="bg-black-800 py-12">
>       <div className="container mx-auto px-4">
>         <div className="flex flex-wrap justify-between">
>           <nav className="w-full lg:w-auto lg:flex-1 lg:text-center flex flex-wrap justify-center">
>             <Link href="/ab" className="footer-link mr-4">
>               <span>Privacy Policy</span>
>             </Link>
>             <Link href="/terms" className="footer-link mr-4">
>               <span>Terms of Service</span>
>             </Link>
>           </nav>
>         </div>
>       </div>
>     </footer>
>   )
> }
> ```
> 
> Componet:
> 
> ```
> import React from 'react'
> 
> export default function ab() {
>   return (
>     <div>
>         <h1>Hello From compoent</h1>
>     </div>
>   )
> }
> ```
> 
> ### Additional information
> ```js
> 404 This page could not be found
> ```
> 
> ### Example
> This is how my files and folder structure look like image = https://i.stack.imgur.com/NrYFg.jpg



> ### Summary
> i am learning nextjs and i am trying to navigate to ab component in nextjs with the link component can anyone can please tell me how to do things properly currently its giving me 404 This page could not be found. error when I try to do that
> 
> Footer:
> 
> ```
> import React from 'react'
> import Link from 'next/link'
> 
> export default function Footer() {
>   return (
>     <footer className="bg-black-800 py-12">
>       <div className="container mx-auto px-4">
>         <div className="flex flex-wrap justify-between">
>           <nav className="w-full lg:w-auto lg:flex-1 lg:text-center flex flex-wrap justify-center">
>             <Link href="/ab" className="footer-link mr-4">
>               <span>Privacy Policy</span>
>             </Link>
>             <Link href="/terms" className="footer-link mr-4">
>               <span>Terms of Service</span>
>             </Link>
>           </nav>
>         </div>
>       </div>
>     </footer>
>   )
> }
> ```
> 
> Componet:
> 
> ```
> import React from 'react'
> 
> export default function ab() {
>   return (
>     <div>
>         <h1>Hello From compoent</h1>
>     </div>
>   )
> }
> ```
> 
> ### Additional information
> ```js
> 404 This page could not be found
> ```
> 
> ### Example
> This is how my files and folder structure look like image = https://i.stack.imgur.com/NrYFg.jpg

Hey there... Did you find a solution? I'm facing the same issue:
